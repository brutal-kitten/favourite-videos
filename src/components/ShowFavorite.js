import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ShowFavorite extends Component {

  static propTypes = {
    showFavorite: PropTypes.func.isRequired
  }

  //calls function that change state in Pagination
  handleClick = (event) => {

    event.preventDefault();
    this.props.showFavorite();
  }

  render() {
    return(
      <div className="showFav" >
        <button tabIndex='0' className="buttonShowFav" type="button" onClick={(event) => this.handleClick(event)}>Show favourite</button>
      </div>
    )
  }
}

export default ShowFavorite;
