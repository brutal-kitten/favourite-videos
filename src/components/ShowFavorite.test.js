import React from 'react';
import ReactDOM from 'react-dom';
import ShowFavorite from './ShowFavorite.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <ShowFavorite />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleClick function after click', () => {
  const mockShowFavorite = jest.fn();
  const props = {
    showFavorite: mockShowFavorite
};

  const wrapper = shallow(<ShowFavorite {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {}
  });

  it('should call showFavorite function', () => {
    expect(mockShowFavorite.mock.calls.length).toBe(1);
  });
})
