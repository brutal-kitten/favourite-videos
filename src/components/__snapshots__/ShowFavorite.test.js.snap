// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="showFav"
>
  <button
    className="buttonShowFav"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    Show favorite
  </button>
</div>
`;
